ddtrace/tracer/option.go

```diff
@@ -266,8 +266,13 @@ func newConfig(opts ...StartOption) *config {
 		c.transport = newHTTPTransport(c.agentAddr, c.httpClient)
 	}
 	if c.propagator == nil {
+		// Max tags length of 512 characters to prevent HTTP 413 issues
+		maxTagsLen := clamp(0, 512, internal.IntEnv(
+			"DD_TRACE_X_DATADOG_TAGS_MAX_LENGTH",
+			defaultMaxTagsHeaderLen,
+		), "DD_TRACE_X_DATADOG_TAGS_MAX_LENGTH")
```

Comment:
```suggestion
		envKey := "DD_TRACE_X_DATADOG_TAGS_MAX_LENGTH"
		max := internal.IntEnv(envKey, defaultMaxTagsHeaderLen)
		if max < 0 {
			log.Warn("Invalid value %d for %s. Setting to 0.", max, envKey)
			max = 0
		}
		if max > 512 {
			log.Warn("Invalid value %d for %s. Maximum allowed is %d. Setting to %d.", max, envKey, 512, 512)
			max = 512
		}
```
A few things:
* I don't think there's any need for a separate `clamp` function. It's easier to inline things.
* Numbers like `512` should be constants.
* If the value of the env. var. is not a number, I think the user should know. The way `internal.IntEnv` is implemented is not that great. If the env. var. is not set, then it makes sense to use the default, but if the env. var. is set and the value is invalid (e.g. not a number), I think the user should be warned.